shared/constants/types/fs.js

```diff
@@ -6,3 +6,21 @@ export type _State = {
 }
 
 export type State = I.RecordOf<_State>
+
+export type FolderVisibility = 'private' | 'public' | 'team' | null
+
+export type FolderHeaderProps = {
+  title: string,
+}
+
+export type FileRowProps = {
+  path: string,
+  icon: IconType,
+  showFileData: () => void,
+}
+
+export type FolderProps = {
+  path: string,
+  visibility: FolderVisibility,
+  items: Array<string>,
```

Comment:
Just a heads up flow does have opaque types which we're starting to use. This allows you to have something that is effectively a string but it treated as an independent type. This allows you to ensure function that take a 'Path' will only pass if a 'Path' is passed and not some arbitrary string.
We don't need to add this complexity now but just pointing out a direction we're going

Reply:
for example https://github.com/keybase/client/blob/41fc5de285d2708841a36012ea8b519bb8048b1b/shared/constants/types/devices.js#L5 . deviceIDs are strings, but we want to treat them as not-strings to get stricter enforcement of types